docs/test-anatomy-sample.md

```diff
@@ -0,0 +1,244 @@
+# Anatomy of a Sample Test
+
+In this example, we will analyze an already existing test on Go for checking
+logging [lim](../tests/lim/). This test returns success when receiving 1 message
+from the log.
+
+## Test Structure
+
+The test directory contains the expected files:
+
+* `Makefile`
+* `eden-config.yml` - eden config file
+* `eden.lim.tests.txt` - test scenario
+* source files for the test binary, in this case, `lim_test.go`
+* subscripts in the `testdata/` directory
+
+## Test Binary
+
+The test binary is written in go, as are most of the integration test.
+
+### Imports
+
+The test relies on processing data structures of which `eden` is aware, and so
+imports the relevant libraries
+[in the go file](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L13-L16):
+
+```go
+import (
+...
+"github.com/lf-edge/eden/pkg/controller/eflowlog"
+"github.com/lf-edge/eden/pkg/controller/einfo"
+"github.com/lf-edge/eden/pkg/controller/elog"
+"github.com/lf-edge/eden/pkg/controller/emetric"
+...
+)
+```
+
+### Main Function
+
+As the binary is a go test file, with setup and teardown, the main function is
+[TestMain](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L83-L142):
+
+```go
+func TestMain(m *testing.M) {
+    // lots of extra code
+}
+```
+
+You can read more about `TestMain` in the
+[go testing reference documentation](https://pkg.go.dev/testing#hdr-Main), and
+usage for EVE
+[on the EVE wiki](https://wiki.lfedge.org/display/EVE/EVE+Integration+Testing).
+
+### Test Functions
+
+In order to actually execute tests, you need to create test functions. These
+all have the signature `func Test*` (other than `TestMain`, of course).
+The lim testing file has four such functions:
+
+* [TestLog](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L144)
+* [TestInfo](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L185)
+* [TestMetrics](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L226)
+* [TestFlowLog](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L269)
+
+#### Single Function
+
+We will explore a single test function, [`TestLog`](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L144),
+which tests a log.
+
+The function itself is quite simple; it has 3 main parts:
+
+1. In the first part, we check for the correctness of the query and initialize the edgeNode variable.
+2. In the second part, we call the `tc.AddProcLog` function which takes 2 arguments, the first argument takes `edgeNode`, and the second argument takes the `func (log * elog.LogItem)` function (that will get all logs for edgeNode), which we immediately describe it, and further in the body of the function (right after return) a self-calling function is described, which performs our check.
```

Comment:
Can you please add `return non-nil value from such funtion indicates that processing are done`.

Reply:
Sure, although this part was straight copy-paste from the existing docs.

Reply:
On second thought, I am not sure I understand what you are saying. 

> return non-nil value from such function

Which function? The function that is the second argument to `AddProcLog()`?

If so, wouldn't a _nil_ function indicate processing is done without any issues, while _non-nil_ would indicate processing is done but had an error?

Reply:
Correct, the second argument. Actually it is reverse logic here. Nil indicates that we still not end with processing and wait for new values. Non-nil will ends our waiting (not global one, but for particular processing function) with success.

Reply:
OK, I will add it.

Reply:
I added it. I also expanded the documentation I the anatomy/example file to make it much clearer what it is doing. Take a look please.